efficientdet/keras/efficientdet_arch_keras.py

```diff
@@ -534,107 +527,44 @@ def get_config(self):
-
-    self.class_net = ClassNet(**options)
-
-  def call(self, inputs, **kwargs):
-    """Run BoxNet/ClassNetfor.
-
-    Args:
-      inputs: input tensor.
-      **kwargs: other parameters.
-
-    Returns:
-      A tuple (class_outputs, box_outputs) for class/box predictions.
-    """
-
-    class_outputs = self.class_net(
-        inputs, min_level=self.min_level, max_level=self.max_level)
+def build_class_and_box_outputs(feats, config):
+  """Builds box net and class net.
 
-    box_outputs = self.box_net(
-        inputs, min_level=self.min_level, max_level=self.max_level)
-
-    return class_outputs, box_outputs
+  Args:
+   feats: input tensor.
+   config: a dict-like config, including all parameters.
 
-  def get_config(self):
-    base_config = super(BuildClassAndBoxOutputs, self).get_config()
+  Returns:
+   A tuple (class_outputs, box_outputs) for class/box predictions.
+  """
+  num_anchors = len(config.aspect_ratios) * config.num_scales
+  num_filters = config.fpn_num_filters
+  class_outputs = ClassNet(num_classes=config.num_classes,
+                           num_anchors=num_anchors,
+                           num_filters=num_filters,
+                           min_level=config.min_level,
+                           max_level=config.max_level,
+                           is_training=config.is_training_bn,
+                           act_type=config.act_type,
+                           repeats=config.box_class_repeats,
+                           separable_conv=config.separable_conv,
+                           survival_prob=config.survival_prob,
+                           use_tpu=config.use_tpu,
+                           data_format=config.data_format)(feats)
+
+  box_outputs = BoxNet(num_anchors=num_anchors,
+                       num_filters=num_filters,
+                       min_level=config.min_level,
+                       max_level=config.max_level,
+                       is_training=config.is_training_bn,
+                       act_type=config.act_type,
+                       repeats=config.box_class_repeats,
+                       separable_conv=config.separable_conv,
+                       survival_prob=config.survival_prob,
+                       use_tpu=config.use_tpu,
+                       data_format=config.data_format)(feats)
 
-    return {
-        **base_config, 'aspect_ratios': self.aspect_ratios,
-        'num_scales': self.num_scales,
-        'num_classes': self.num_classes,
-        'fpn_num_filters': self.fpn_num_filters,
-        'min_level': self.min_level,
-        'max_level': self.max_level,
-        'is_training_bn': self.is_training_bn,
-        'act_type': self.act_type,
-        'box_class_repeats': self.box_class_repeats,
-        'separable_conv': self.separable_conv,
-        'survival_prob': self.survival_prob,
-        'use_tpu': self.use_tpu,
-        'data_format': self.data_format,
-        **self.args
-    }
+  return class_outputs, box_outputs
```

Comment:
I'm not confident with this change, the goal here with keras is to create a tf.keras.model wich can use all features from tf and keras, and be heavily customizable.

Reply:
You have break variable names and these layers are useless.

Reply:
All weights was good so this was not critical as it was still possible to load model

Reply:
Maybe you could just do your changes inside the layer ?